site/content/docs/user/known-issues.md

```diff
@@ -306,6 +313,34 @@ your workloads inside the cluster via the nodes IPv6 addresses.
 
 See Previous Discussion: [kind#1326]
 
+## Failed to get rootfs info / "stat failed on /dev/..."
+
+On some systems, creating a cluster times out with these errors in kubelet.log (device varies):
+
+```
+stat failed on /dev/nvme0n1p3 with error: no such file or directory
+"Failed to start ContainerManager" err="failed to get rootfs info: failed to get device for dir \"/var/lib/kubelet\": could not find device with major: 0, minor: 40 in cached partitions map"
+```
+
+Kubernetes needs access to storage device nodes in order to do some stuff, e.g. tracking free disk space. Therefore, Kind needs to mount the necessary device nodes from the host into the control-plane container — however, it cannot always determine which device Kubernetes requires, since this varies with the host filesystem. For example, Kind doesn't handle BTRFS, which is the default for modern Fedora.
+
+This can be worked around by including the necessary device as an extra mount in the cluster configuration file.
+
+```yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraMounts:
+    - hostPath: /dev/nvme0n1p3
+      containerPath: /dev/nvme0n1p3
```

Comment:
should we add some instructions for users on how to obtain the device path?

Reply:
That's in the following paragraph, where I state "the expected device is named in the error message".

Reply:
Just in case, in my Fedora 35 it complained about `/dev/mapper/luks-903aad3d-...` and using `hostPath/containerPath: /dev/mapper/luks-903aad3d-..` didn't worked because it is a symlink to `/dev/dm-0`. Using `/dev/dm-0` worked.

```
ls -l /dev/mapper/
total 0
crw-------. 1 root root 10, 236 ene 11 08:06 control
lrwxrwxrwx. 1 root root       7 ene 11 08:06 luks-903aad3d-... -> ../dm-0
```

Reply:
So, two variations - if the device cited in the error message is under `/dev/mapper`, the extra mount should be the /dev/dm-* device it points to; otherwise it should be the device from the message?

Reply:
I guess it needs to be the real device, not a symlink. So if the error complains about `/dev/foo/bar`, the config needs to have the real device such as `readlink -f /dev/foo/bar`

```
MYDEVICE=$(readlink -f /dev/mapper/luks-903aad3d-...)
cat << EOF | kind create cluster --config=-
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  extraMounts:
  - hostPath: ${MYDEVICE}
    containerPath: ${MYDEVICE}
    propagation: HostToContainer
EOF
```